### DISCUSSION TITLE
Compiling mediamtx without the embedded libcamera

### DISCUSSION THREAD

**Discussion by bdcleret** [bluenviron/mediamtx#3774] [Questions] [answered]
### Question

Hi,

I try to use mediamtx with an arducam pivariety camera that comes with its own custom libcamera.
I noted that I am supposed to compile mediamtx from source for this: 

"Be aware that the server is not compatible with cameras that requires a custom libcamera (like some ArduCam products), since it comes with a bundled libcamera. If you want to use a custom one, you can [compile from source](https://github.com/bluenviron/mediamtx#compile-from-source)."

However when I compile from (source following the instructions ("standard" section, I cloned the repo yesterday), I end up with a mediamtx binary that apparently still integrates and uses its embedded libcamera.

```
Sep 15 21:51:08 mediamtx[117268]: [58:06:22.723040225] [117269]  WARN CameraSensorProperties camera_sensor_properties.cpp:286 No static properties available for 'arducam-pivariety'
Sep 15 21:51:08 mediamtx[117268]: [58:06:22.723141595] [117269]  WARN CameraSensorProperties camera_sensor_properties.cpp:288 Please consider updating the camera sensor properties database
Sep 15 21:51:08 mediamtx[117268]: [58:06:22.730282872] [117269] ERROR IPAProxy ipa_proxy.cpp:149 Configuration file 'arducam-pivariety.json' not found for IPA module 'rpi/vc4'
Sep 15 21:51:08 mediamtx[117268]: [58:06:22.735036143] [117269] ERROR IPARPI ipa_base.cpp:125 Could not create camera helper for arducam-pivariety
Sep 15 21:51:08 mediamtx[117268]: [58:06:22.735106235] [117269] ERROR RPI pipeline_base.cpp:798 Failed to load a suitable IPA library
Sep 15 21:51:08 mediamtx[117268]: [58:06:22.735123291] [117269] ERROR RPI vc4.cpp:215 Failed to register camera arducam-pivariety 10-000c: -22
Sep 15 21:51:08 mediamtx[8714]: 2024/09/15 21:51:08 ERR [path cam] [RPI Camera source] camera_create(): selected camera is not available
```

Are there any specific compilation flags/settings/steps that would allow to use an external libcamera?

**Comment by aler9** [comment, marked as answer]
Hello, yes, that procedure was recently updated and the README was not. You have to download [mediamtx-rpicamera](https://github.com/bluenviron/mediamtx-rpicamera) and follow instructions here:

https://github.com/bluenviron/mediamtx-rpicamera?tab=readme-ov-file#compile-against-an-external-libcamera

**Reply by bdcleret** [reply]
Thanks for the quick answer.
I followed the steps you provided, it is working as expected now.
Thanks again!

**Comment by DirtyOptics** [comment]
Sorry to re-invigorate an old thread, But I am also trying to get mediamtx to work with an arducam IMX462 module using the pivariety libcamera drivers. 

I can get the camera to run no dramas if running libcamera manually, then I tried adding something like this to mediamtx.yml

```
paths:
  cam:
    runOnDemand: libcamera-vid -t 0 --width 640 --height 480 --framerate 15 --inline --codec h264 -o -
    runOnDemandRestart: yes 
```

However this probably isnt the right approach.....Any advice?

**Reply by aler9** [reply]
@DirtyOptics please read here: https://github.com/bluenviron/mediamtx?tab=readme-ov-file#custom-libcamera

If you have further questions, open a dedicated discussion.
